### ISSUE TITLE
`pre-start` script in `blobstore` job can take a long time

### ISSUE THREAD

**Issue by nikhilsuvarna** [cloudfoundry/cloud_controller_ng#1013]
Thanks for submitting an issue to `cloud_controller_ng`. We are always trying to improve! To help us, please fill out the following template.

## Issue

In large environments, recreating of VM that runs the`blobstore` job using `bosh recreate` can take a long time during the `pre-start` script execution, specifically around `chown -R`. This can cause synchronization issues with bosh and monit

## Context

cf-release v252

While running `monit start all`, the `pre-start` script can take a long time to run causing timing issues with the execution of the script and monit timeouts causing bosh to fail the job. 

`ps -ef` revealed that `chown -R` is taking a long time process a large number of files. 

```
950 ? S 0:00 | \_ svlogd -tt /var/vcap/bosh/log 
951 ? S<l 0:25 | \_ /var/vcap/bosh/bin/bosh-agent -P ubuntu -C /var/vcap/bosh/agent.json 
7621 ? S< 0:00 | \_ bash /var/vcap/jobs/blobstore/bin/pre-start 
7639 ? R< 3:17 | \_ chown -R vcap:vcap /var/vcap/sys/run/blobstore /var/vcap/sys/log/blobstore /var/vcap/store/shared /var/vcap/store/shared/tmp/uploads /var/vcap/packages/nginx_webdav /var/vcap/packages/nginx_webdav/.. 
```
In this case, the total number of files is `2161738`

upon reviewing `pre-start` script

```
 #!/usr/bin/env bash

set -ex

function setup_blobstore_directories { 
local run_dir=/var/vcap/sys/run/blobstore 
local log_dir=/var/vcap/sys/log/blobstore 
local data=/var/vcap/store/shared 
local tmp_dir=$data/tmp/uploads 
local nginx_webdav_dir=/var/vcap/packages/nginx_webdav

mkdir -p $run_dir 
mkdir -p $log_dir 
mkdir -p $data 
mkdir -p $tmp_dir 
chown -R vcap:vcap $run_dir $log_dir $data $tmp_dir $nginx_webdav_dir "${nginx_webdav_dir}/.." 
}
```

If there is a concern around the file ownership outside of `mkdir -p`,  the suggestion is  that it should be addressed elsewhere. 

## Expected result

`bosh recreate` should complete in a reasonable time within `monit`'s timeout values. 

## Current result

`pre-start` for `nfs-server` jobs takes a long time while processing over 2 million files. 

## Possible Fix

Use alternative methods to check for `vcap` ownership rather than `chown -R` across entire directories each time a `monit start` is executed.

**Comment by lisamcho** [comment]
Thanks to James Wynne for a snippet with benchmarking data...
slack conversation here (https://pivotal.slack.com/archives/C3LV25ZCM/p1511889152000134)
```
~ $ time sudo find /srv/backup/me -type f | wc -l
843004

real    0m2.102s
user    0m0.911s
sys     0m1.508s
~ $ time sudo find /srv/backup/me -type d | wc -l
31552

real    0m1.617s
user    0m0.554s
sys     0m1.077s

~ $ time sudo chmod -R 700 /srv/backup/me

real    0m51.598s
user    0m0.962s
sys     0m17.339s

~ $ time sudo find /srv/backup/me -exec chmod 700 '{}' \;

real    30m32.551s
user    2m23.450s
sys     22m22.665s
~ $ time sudo find /srv/backup/me -type d -exec chmod 700 '{}' \;

real    1m9.317s
user    0m6.608s
sys     0m50.764s
```

**Comment by jloar-pivotal** [comment]
The customer that reported this issue is hoping we can implement this fix and get it into the latest PCF 1.12 release. Is it possible to get this small change worked in?

Thanks!

**Comment by zrob** [comment]
will be available in capi release 1.6.1.0
